### ISSUE TITLE
Issue with infinite wave

### ISSUE THREAD

**Issue by abhiturbo** [JoshuaKGoldberg/TypeStat#1054]
### 🐛 Bug Report

-   TypeStat version: 0.6.1
-   TypeScript version: 4.5.4 and 4.8.4
-   Node version: 16.x and 18.x

#### Actual Behavior

After running `npx typestat --config typestat.json` the wave runs endless for more than hours and at last I had to quit the process.

#### Expected Behavior

We should get an info for estimated time left or percentage of progress left to complete the task.

#### Reproduction

Run `npx typestat --config typestat.json` on my React UI library.

cc: Hi @JoshuaKGoldberg , could you plz. guide me to fix the above issue or is something I am doing wrong in the process of running typestat.

**Comment by abhiturbo** [comment]
Hi @JoshuaKGoldberg, Thanks for the response!
Ex. Repo: https://github.com/OutpostHQ/jengaui I did tried in my private repo as well and got the same response. The wave went 1000+ in 1+ hrs. so lastly I had to quit the process.
```
// typestat.json

[
    {
        "fixes": {
            "incompleteTypes": true,
            "noImplicitAny": true,
            "inferableTypes": true,
            "noImplicitThis": true,
            "strictNonNullAssertions": true
        },
        "include": [
            "test/**/*.{ts,tsx}"
        ],
        "types": {
            "strictNullChecks": true
        },
        "projectPath": "./packages/shared-tsconfig/react-library.json"
    },
    {
        "exclude": [
            "test/**/*.{ts,tsx}"
        ],
        "fixes": {
            "incompleteTypes": true,
            "noImplicitAny": true,
            "inferableTypes": true,
            "noImplicitThis": true
        },
        "include": [
            "./packages/**/*.{ts,tsx}"
        ],
        "types": {
            "strictNullChecks": true
        },
        "projectPath": "./packages/shared-tsconfig/react-library.json"
    },
    {
        "fixes": {
            "incompleteTypes": true,
            "noImplicitAny": true,
            "inferableTypes": true,
            "noImplicitThis": true
        },
        "include": [
            "test/**/*.{ts,tsx}",
            "./packages/**/*.{ts,tsx}"
        ],
        "types": {
            "strictNullChecks": true
        },
        "projectPath": "./packages/shared-tsconfig/react-library.json"
    }
]
```
![image](https://user-images.githubusercontent.com/94604790/194053041-506ece10-b12b-4a11-99aa-c8a42d354a72.png)

**Comment by JoshuaKGoldberg** [comment]
🤔 when I try to use that `typestat.json` with the repo as-is, I get a different error. 

```shell
git clone https://github.com/JoshuaKGoldberg/jengaui/tree
cd jengaui
git checkout e117755d869740c923c8183e3ca744c9f632d47d
npx typestat -c typestat.json
```

```plaintext
🚀 Welcome to TypeStat!
TypeStat will run through the 3 options objects specified in typestat.json to modify your source code.
This may take a while...


Starting options object 1 of 3.

Starting the core mutation engine. This terminal will log whenever a "wave" of mutations are written to files.
Core mutations will complete when two waves pass with no mutations.
The following fixes will be applied:
        * incompleteTypes
        * noImplicitAny
        * noImplicitThis
        * strictNonNullAssertions
        * inferableTypes

Starting wave 1...
Completed wave 1. Wrote mutations to 0 files.
Done.

Finished options object 1 of 3.

Error: Could not run options object 2: At least one path including node_modules was included implicitly: '/Users/josh/repos/jengaui/packages/button/node_modules/@react-types/button/src/index.d.ts'. Either adjust TypeStat's included files to not include node_modules (recommended) or explicitly include node_modules/ (not recommended).
```

Could you please post an exact set of steps to get the behavior you're describing?

**Comment by JoshuaKGoldberg** [comment]
@abhiturbo I just added infinite wave detection and an assortment of bug fixes in version 0.6.2. Could you try again with `npx typestat@0.6.2`? I have a hope your issue should be fixed - or at least, the CLI should bail out after much less time than an hour!

**Comment by JoshuaKGoldberg** [comment]
I'm going to close this as non-actionable, as it's been a bit over a month. Thanks for filing - and let me know if there's anything else I can help with @abhiturbo!
